src/Configuration/Configuration.php

```diff
@@ -0,0 +1,243 @@
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Configuration;
+
+use Infection\Configuration\Entry\Logs;
+use Infection\Configuration\Entry\PhpUnit;
+use Infection\Configuration\Entry\Source;
+use Webmozart\Assert\Assert;
+
+/**
+ * @internal
+ * @final
+ */
+class Configuration
+{
+    private const LOG_VERBOSITY = [
+        'all',
+        'none',
+        'default',
+    ];
+
+    private const TEST_FRAMEWORKS = [
+        'phpunit',
+        'phpspec',
+    ];
+
+    private const FORMATTER = [
+        'dot',
+        'progress',
+    ];
+
+    private $timeout;
+    private $source;
+    private $logs;
+    private $logVerbosity;
+    private $tmpDir;
+    private $phpUnit;
+    private $mutators;
+    private $testFramework;
+    private $bootstrap;
+    private $initialTestsPhpOptions;
+    private $testFrameworkOptions;
+    private $existingCoveragePath;
+    private $debug;
+    private $onlyCovered;
+    private $formatter;
+    private $noProgress;
+    private $ignoreMsiWithNoMutations;
+    private $minMsi;
+    private $showMutations;
+    private $minCoveredMsi;
+    private $stringMutators;
+
+    public function __construct(
+        ?int $timeout,
+        Source $source,
+        Logs $logs,
+        string $logVerbosity,
+        ?string $tmpDir,
+        PhpUnit $phpUnit,
+        array $mutators,
+        ?string $testFramework,
+        ?string $bootstrap,
+        ?string $initialTestsPhpOptions,
+        ?string $testFrameworkOptions,
+        ?string $existingCoveragePath,
+        bool $debug,
+        bool $onlyCovered,
+        string $formatter,
+        bool $noProgress,
+        bool $ignoreMsiWithNoMutations,
+        ?float $minMsi,
+        bool $showMutations,
+        ?float $minCoveredMsi,
+        ?string $stringMutators
+    ) {
+        Assert::nullOrGreaterThanEq($timeout, 1);
```

Comment:
Those are "harsh" errors: they will end up in a straight up failure. I think however that if we want to avoid them, this should be done downstream either by validating the configuration file (which we already do, so it would highlight a missing validation rule from the `schema.json`, or the input)